lightly/api/api_workflow_collaboration.py

```diff
@@ -0,0 +1,76 @@
+from os import access
+from typing import List
+
+from lightly.openapi_generated.swagger_client.models.create_entity_response import CreateEntityResponse
+from lightly.openapi_generated.swagger_client.models.dataset_create_request import DatasetCreateRequest
+
+from lightly.openapi_generated.swagger_client.models.shared_access_config_create_request import SharedAccessConfigCreateRequest
+from lightly.openapi_generated.swagger_client.models.shared_access_config_data import SharedAccessConfigData
+from lightly.openapi_generated.swagger_client.models.shared_access_type import SharedAccessType
+
+from lightly.openapi_generated.swagger_client.models.dataset_data import DatasetData
+from lightly.openapi_generated.swagger_client.rest import ApiException
+from lightly.openapi_generated.swagger_client.models.dataset_type import DatasetType
+
+class _CollaborationMixin:
+
+    def share_dataset(self, dataset_id: str, user_emails: List[str]):
+        """Shares dataset with a list of users
+
+        This method overwrites the list of users that have had access to the dataset
+        before. If you want to add someone new to the list make sure you get the
+        list of users with access beforehand and add them as well.
+
+        Args:
+          dataset_id:
+            Identifier of dataset
+          user_emails:  
+            List of email addresses of users to grant write permission
+
+        Examples:
+          >>> # share a dataset with an user
+          >>> client = ApiWorkflowClient(token="MY_AWESOME_TOKEN")
+          >>> client.share_dataset(dataset_id="MY_DATASET_ID", user_emails=["user@something.com"])
+          >>>
+          >>> # share dataset with a user while keep sharing it with previous users
+          >>> client = ApiWorkflowClient(token="MY_AWESOME_TOKEN")
+          >>> user_emails = client.get_shared_users(dataset_id="MY_DATASET_ID")
+          >>> user_emails.extend(["additional_user2@something.com"])
+          >>> client.share_dataset(dataset_id="MY_DATASET_ID", user_emails=user_emails)
+          >>>
+          >>> # revoke access to all users
+          >>> client = ApiWorkflowClient(token="MY_AWESOME_TOKEN")
+          >>> user_emails = client.share_dataset(dataset_id="MY_DATASET_ID", user_emails=[])
+        """
+        body = SharedAccessConfigCreateRequest(access_type=SharedAccessType.WRITE, users=user_emails)
+        self._collaboration_api.create_or_update_shared_access_config_by_dataset_id(body=body, dataset_id=dataset_id)
+
+
+    def get_shared_users(self, dataset_id: str) -> List[str]:
```

Comment:
for future safety, allow one to possible to pass access_type but default to SharedAccessType.WRITE

Reply:
I thought about that as well but then decided against it. I'd rather update the API specs as there is no use case for now to differentiate between read and write permission.

Reply:
changing the API specs is a breaking change. Then lets just pseudo default to write perm

Reply:
pseudo default --> what is in the PR or adding the default argument to use WRITE permission?